### PR TITLE
Process AMMM at startup in bios handler

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The commit make changes to process Auto Memory Mirror Mode. If the value found in VPD is default, BIOS value is backed up in VPD else VPD is restored into BIOS table.

This is required to sync the data between BIOS and VPD once PLDM service is running.

### CODE REVIEW

File: src/bios_handler.cpp [hunk 1]
```diff
@@ -143,6 +143,9 @@ void IbmBiosHandler::backUpOrRestoreBiosAttributes()
 {
     // process FCO
     processFieldCoreOverride();
+
+    // process memory mirror mode
+    processMemoryMirrorMode();
```
Comment (jinuthomas) [inline src/bios_handler.cpp:148]: processAutoMemoryMirror
as the keyword is AMM like FCO

File: src/bios_handler.cpp [hunk 2]
```diff
@@ -143,6 +143,9 @@ void IbmBiosHandler::backUpOrRestoreBiosAttributes()
 {
     // process FCO
     processFieldCoreOverride();
+
+    // process memory mirror mode
```
Comment (jinuthomas) [inline src/bios_handler.cpp:147]: keep it as above 
// process AMM

File: src/bios_handler.cpp [hunk 3]
```diff
@@ -215,4 +218,42 @@ void IbmBiosHandler::saveFcoToVpd(int64_t i_fcoInBios)
...
+
+void IbmBiosHandler::processMemoryMirrorMode()
+{
+    auto l_kwdValueVariant = dbusUtility::readDbusProperty(
+        constants::pimServiceName, constants::systemVpdInvPath,
+        constants::utilInf, constants::kwdAMM);
+
+    if (auto pVal = std::get_if<std::string>(&l_kwdValueVariant))
+    {
+        auto l_ammmValInVpd = *pVal;
```
Comment (jinuthomas) [inline src/bios_handler.cpp:230]: nit : amm

File: src/bios_handler.cpp [hunk 4]
```diff
@@ -215,4 +218,42 @@ void IbmBiosHandler::saveFcoToVpd(int64_t i_fcoInBios)
...
+
+        // Check if memory mirror mode value is default in VPD.
```
Comment (jinuthomas) [inline src/bios_handler.cpp:232]: nit: auto memory mirror

File: src/bios_handler.cpp [hunk 5]
```diff
@@ -215,4 +218,42 @@ void IbmBiosHandler::saveFcoToVpd(int64_t i_fcoInBios)
...
+                return;
+            }
+
+            logging::logMessage(
+                "Invalid type recieved for auto memory mirror mode from BIOS.");
+        }
+        return;
+    }
+    logging::logMessage(
+        "Invalid type recieved for auto memory mirror mode from VPD.");
```
Comment (jinuthomas) [inline src/bios_handler.cpp:257]: nit: value not type

File: src/bios_handler.cpp [hunk 6]
```diff
@@ -215,4 +218,42 @@ void IbmBiosHandler::saveFcoToVpd(int64_t i_fcoInBios)
...
+            {
+                std::string l_ammmInBios = *pVal;
+                (void)l_ammmInBios;
+                // TODO: save to BIOS.
+
+                return;
+            }
+
+            logging::logMessage(
+                "Invalid type recieved for auto memory mirror mode from BIOS.");
```
Comment (jinuthomas) [inline src/bios_handler.cpp:252]: nit: value not type

**Review by jinuthomas — CHANGES_REQUESTED** [review]

File: src/bios_handler.cpp (hunk 6 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp:252]: We are doing a type check there for std::string

File: src/bios_handler.cpp (hunk 5 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp:257]: Same here, doing a type check for std::string.

File: src/bios_handler.cpp (hunk 2 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp]: Done.

File: src/bios_handler.cpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp]: Done.

File: src/bios_handler.cpp (hunk 3 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp]: Done.

File: src/bios_handler.cpp (hunk 4 above)
Comment (SunnySrivastava1984) [reply src/bios_handler.cpp]: Done.

File: src/bios_handler.cpp (hunk 6 above)
Comment (jinuthomas) [reply src/bios_handler.cpp:267]: ok

File: src/bios_handler.cpp (hunk 5 above)
Comment (jinuthomas) [reply src/bios_handler.cpp:272]: ok

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**